pylint/checkers/refactoring/refactoring_checker.py

```diff
@@ -2199,3 +2204,43 @@ def _check_unnecessary_list_index_lookup(
                 args=(node.target.elts[1].name,),
                 confidence=HIGH,
             )
+
+    def _enumerate_with_start(self, node: nodes.For | nodes.Comprehension) -> bool:
+        """Check presence of `start` kwarg or second argument to enumerate.
+
+        For example:
+
+        `enumerate([1,2,3], start=1)`
+        `enumerate([1,2,3], 1)`
+
+        If `start` is assigned to `0`, the default value, this is equivalent to
+        not calling `enumerate` with start.
+        """
+        if len(node.iter.args) > 1:
+            # We assume the second argument to `enumerate` is the `start` int arg.
+            # It's a reasonable assumption for now as it's the only possible argument:
+            # https://docs.python.org/3/library/functions.html#enumerate
+            start_arg = node.iter.args[1]
+            start_val = self._get_start_value(start_arg)
+            if start_val is None:
+                return False
+            return not start_val == 0
+
+        for keyword in node.iter.keywords:
+            if keyword.arg == "start":
+                start_val = self._get_start_value(keyword.value)
+                if start_val is None:
+                    return False
+                return not start_val == 0
+
+        return False
+
+    def _get_start_value(self, node: nodes.NodeNG) -> Any:
+        if isinstance(node, nodes.Name):
+            inferred = utils.safe_infer(node)
```

Comment:
It seems like the confidence is not HIGH anymore if we're in this branch. Maybe we should return the confidence in this function ?

Reply:
oh good call! So the rule is, anytime inference is used to determine whether a msg should be emitted, then use INFER confidence, correct?

Reply:
>  So the rule is, anytime inference is used to determine whether a msg should be emitted, then use INFER confidence

Yeap !